File: ebs-springboot/src/main/java/com/erp/model/Product.java
package com.erp.model;

import javax.persistence.*;

@Entity(name = "product")
@Table(name = "product")
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(name = "product_code")
    private Long productCode;

    @Column(name = "product_name")
    private String productName;

    @Column(name = "product_category_code")
    private Long productCategoryCode;

    @Column(name = "product_category_name")
    private String productCategoryName;    
    
    @Column(name = "description")
    private String description;
    
    @Column(name = "reorder_level")
    private String reorderLevel;

    @Column(name = "status")
    private String status;

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public Long getProductCode() {
		return productCode;
	}

	public void setProductCode(Long productCode) {
		this.productCode = productCode;
	}

	public String getProductName() {
		return productName;
	}

	public void setProductName(String productName) {
		this.productName = productName;
	}

	public Long getProductCategoryCode() {
		return productCategoryCode;
	}

	public void setProductCategoryCode(Long productCategoryCode) {
		this.productCategoryCode = productCategoryCode;
	}

	public String getProductCategoryName() {
		return productCategoryName;
	}

	public void setProductCategoryName(String productCategoryName) {
		this.productCategoryName = productCategoryName;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public String getReorderLevel() {
		return reorderLevel;
	}

	public void setReorderLevel(String reorderLevel) {
		this.reorderLevel = reorderLevel;
	}

	public String getStatus() {
		return status;
	}

	public void setStatus(String status) {
		this.status = status;
	}

}

File: ebs_flutter/lib/model/Product.dart
class Product {
  final int? id;
  final int? productCode;
  final String? productName;
  final int? productCategoryCode;
  final String? productCategoryName;
  final String? description;
  final String? reorderLevel;
  final String? status;

  Product({
    this.id,
    this.productCode,
    this.productName,
    this.productCategoryCode,
    this.productCategoryName,
    this.description,
    this.reorderLevel,
    this.status,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['id'],
      productCode: json['productCode'],
      productName: json['productName'],
      productCategoryCode: json['productCategoryCode'],
      productCategoryName: json['productCategoryName'],
      description: json['description'],
      reorderLevel: json['reorderLevel'],
      status: json['status'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'productCode': productCode,
      'productName': productName,
      'productCategoryCode': productCategoryCode,
      'productCategoryName': productCategoryName,
      'description': description,
      'reorderLevel': reorderLevel,
      'status': status,
    };
  }
}